packages/flutter/lib/src/widgets/raster_widget.dart

```diff
@@ -0,0 +1,165 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'media_query.dart';
+
+/// A widget that replaces its child with a rasterized version of said child.
+class RasterWidget extends SingleChildRenderObjectWidget {
```

Comment:
@goderbauer @dnfield here is my next attempt at a reusable widget for this, which does only one thing. some notes:

1. Currently catches PictureRasterizationException. We just need some way of detecting tester + software backend so that I can fall back to better behavior.
2. I'm using a bool for raster vs non-raster, but I don't have a good way of saying "keep using the raster but re-create the raster"